python/cudf/cudf/core/series.py

```diff
@@ -1250,17 +1250,85 @@ def nans_to_nulls(self):
         else:
             return self
 
-    def all(self, axis=0, skipna=True, level=None):
+    def all(self, axis=0, bool_only=None, skipna=True, level=None, **kwargs):
         """
+        Return whether all elements are True in Series.
+
+        Parameters
+        ----------
+
+        skipna: bool, default True
+            Exclude NA/null values. If the entire row/column is NA and
+            skipna is True, then the result will be True, as for an
+            empty row/column.
+            If skipna is False, then NA are treated as True, because
+            these are not equal to zero.
+
+        Returns
+        -------
+        scalar
+
+        Notes
+        -----
+        Parameters currently not supported are `axis`, `bool_only`, `level`.
+
+        Examples
+        --------
+        >>> import cudf
+        >>> ser = cudf.Series([1, 5, 2, 4, 3])
+        >>> ser.all()
+        True
         """
-        assert axis in (None, 0) and skipna is True and level in (None,)
-        return self._column.all()
 
-    def any(self, axis=0, skipna=True, level=None):
+        assert axis in (None, 0) and level in (None,)
```

Comment:
Validate the `bool_only` flag? Also, we shouldn't be using assertions for this as that will raise an `AssertionError` instead of a `NotImplementedError`

Reply:
Yep, sure 👍